lib/executor.js

```diff
@@ -175,6 +175,7 @@ function setupDataSources(app, instructions) {
       useEnvVars: true,
     };
     obj = getUpdatedConfigObject(app, obj, opts);
+    obj.lazyConnect = true;
```

Comment:
I tried with local db, I think the `connect` will be triggered when data query happens, like hitting endpoint from Explorer. And also be triggered when [`datasource.ping()`](https://github.com/strongloop/loopback-connector-mongodb/blob/master/lib/mongodb.js#L1131) is called.